### DISCUSSION TITLE
Vertical alignment in tables

### DISCUSSION THREAD

**Discussion by leomayer** [lvjr/tabularray#125] [Q&A]
If I follow the documentation of `1.3. Cell alignment` and the definitions of `Table 2.4` the vertical alignment could be set with `m` to be vertically centered.

The following code:

```latex
\begin{table}[!htpb]
\centering
\caption{Table Caption}
\begin{tblr}{
		colspec = {|[red3,1pt]X[l,2]|[red3,1pt]X[r]|X[r]|[red3,1pt]X[r]|X[r,m]|[red3,1pt]},
		row{1,2} = {c,azure8,font=\bfseries},
		column{1} = {purple8,font=\bfseries},
		cell{1}{1} = {r=2}{m,white}, %multirow
		cell{1}{2,4} = {c=2}{c}, % multicolumn,
	}
	\hline[1pt,blue3]
	& 2019  &      & 2020   &      \\
	\hline
	& win   & loss & win    & loss \\ 
	\hline[0.7pt]
	Aggressive   & Typography  & 15\% &   21\% & 25\% \\ 
	\hline
	Moderate     & 7.5\% & 11\% & 11.5\% & 23\% \\ 
	\hline
	Conservative & 8.5\% & 12\% & 8.24\% & 12\% \\
	\hline[1pt,blue3]
\end{tblr}
\end{table}
```

produces
![Bildschirmfoto von 2021-10-01 12-56-07](https://user-images.githubusercontent.com/2641128/135608832-29c8a7f2-6d6f-4e3a-b303-b658b3d68b53.png)

I would have expected that the parameter `X[r,m]` would align the content in the middle. But  as depicted the last row remains on the top. When I set `m` in the column with `Typography` this works as expected.

I would guess this is a bug but don't know if it works as designed or if the settings should be used that way.

**Comment by lvjr** [comment]
When there is only one line in the cell, `m`, `t`, and `b` alignments are the same. The same is true for `tabular` environment.

**Comment by mxa** [comment]
Visually the text seems to stick to the top of the cell in any case. That's especially true if there are no pqjyg letters in the text. Is there a way to manipulate the vertical alignment?
Looking at the lines with 2019, 2020, win, loss above this is quite apparent.

**Reply by lvjr** [reply]
you can use `stretch=0` to remove the default struts.

**Reply by mxa** [reply]
Cool. Actually `stretch=1.5` had the desired effect

**Comment by leomayer** [comment]
Just a stupid question: is the parameter dependent on the number of lines? E.g with 3 or 4 lines the value need to be adapted?

**Reply by lvjr** [reply]
Yes, positive `stretch` is not fit for all cases. Therefore it would be better to use `stretch=0` with a larger `rowsep`.
